student/src/main/java/tds/student/services/remote/RemoteOpportunityService.java

```diff
@@ -268,9 +272,12 @@ public TestConfig startTest(final OpportunityInstance oppInstance, final String
     if (!isRemoteExamCallsEnabled) {
       return testConfig;
     }
-  
+
+    final HttpServletRequest httpRequest = HttpContext.getCurrentContext().getRequest();
```

Comment:
Can we do this at the controller layer and modify the `OpportunityInstance` to have this value?  HttpRequest is specific to the endpoint.  This will blow up if this service would ever be called from a non Http flow, correct?